src/PowerShellEditorServices/Services/DebugAdapter/DebugService.cs

```diff
@@ -687,8 +680,20 @@ private async Task FetchGlobalAndScriptVariablesAsync()
             var scopeVariableContainer = new VariableContainerDetails(this.nextVariableId++, "Scope: " + scope);
             this.variables.Add(scopeVariableContainer);
 
-            IReadOnlyList<PSObject> results = await _executionService.ExecutePSCommandAsync<PSObject>(psCommand, CancellationToken.None)
-                .ConfigureAwait(false);
+            IReadOnlyList<PSObject> results;
+            try
+            {
+                results = await _executionService.ExecutePSCommandAsync<PSObject>(psCommand, CancellationToken.None)
+                    .ConfigureAwait(false);
+            }
+            catch (CmdletInvocationException ex)
+            {
+                if (!ex.ErrorRecord.CategoryInfo.Reason.Equals("PSArgumentOutOfRangeException"))
```

Comment:
```suggestion
                // It's possible to be asked to run `Get-Variable -Scope N` where N is a number that
                // exceeds the available scopes. In this case, the command throws this exception,
                // but there's nothing we can do about it and we shouldn't crash the debugger, so we
                // just return no results instead. All other exceptions should be thrown again.
                if (!ex.ErrorRecord.CategoryInfo.Reason.Equals("PSArgumentOutOfRangeException"))
```
Suggested comment to resolve Paul's request.